### PR TITLE
Pgcraft misc v2

### PR DESCRIPTION
**Pull request by mbouron-gpsw**

### CODE REVIEW

File: tests/data.py [hunk 1]
```diff
@@ -532,3 +530,15 @@ def data_vertex_and_fragment_blocks(cfg):
...
+@test_cuepoints(points={'c': (0, 0)}, nb_keyframes=1, tolerance=1)
+@scene()
+def data_vertex_and_fragment_blocks(cfg):
+    return _data_vertex_and_fragment_blocks(cfg)
+
+
+@test_cuepoints(points={'c': (0, 0)}, nb_keyframes=1, tolerance=1)
+@scene()
+def data_vertex_and_fragment_blocks_std430(cfg):
+    return _data_vertex_and_fragment_blocks(cfg, True)
```
Comment (cboesch-gpsw) [inline tests/data.py:544]: Can you explicitly send the layout instead, and do it in both calls?
```python
def _data_vertex_and_fragment_blocks(cfg, layout):
    ...


def data_vertex_and_fragment_blocks(cfg):
    return _data_vertex_and_fragment_blocks(cfg, layout='std140')

...

def data_vertex_and_fragment_blocks_std430(cfg):
    return _data_vertex_and_fragment_blocks(cfg, layout='std430')
```

File: libnodegl/pgcraft.c [hunk 2]
```diff
@@ -168,6 +168,19 @@ static const char *get_glsl_type(int type)
     return ret;
 }
 
+static int get_next_available_binding(struct pgcraft *s, int stage, int type)
```
Comment (cboesch-gpsw) [inline libnodegl/pgcraft.c:171]: Not a fan of using the `get_*` semantic for functions altering the state. Maybe `query_next_binding()` or `request_next_binding()`

File: libnodegl/pipeline.h [hunk 3]
```diff
@@ -101,6 +97,13 @@ struct pipeline_params {
     int nb_attributes;
 };
 
+struct pipeline_params {
+    int type;
+    const struct pipeline_graphics graphics;
+    const struct program *program;
+    struct pipeline_layout layout;
```
Comment (cboesch-gpsw) [inline libnodegl/pipeline.h]: In the commit message ("pipeline: add the layout fields to a dedicated pipeline_layout structure"), can you replace the "add" with "move"?

**Review by cboesch-gpsw — CHANGES_REQUESTED** [review]
LGTM aside from the cosmetics

File: libnodegl/pipeline.h (hunk 3 above)
Comment (mbouron-gpsw) [reply libnodegl/pipeline.h]: Done.

File: tests/data.py (hunk 1 above)
Comment (mbouron-gpsw) [reply tests/data.py]: Done.

File: libnodegl/pgcraft.c (hunk 2 above)
Comment (mbouron-gpsw) [reply libnodegl/pgcraft.c]: Renamed to `request_next_binding()`

**Review by cboesch-gpsw — APPROVED** [review]

**Pull request merged**